### ISSUE TITLE
Fill part of form with code

### ISSUE THREAD

**Issue by Richard87** [wittydeveloper/react-apollo-form#13]
Hi!

I have a required `clientMutationId` that I want to fill programatically, is there a simple way of forcing `ApplicationForm` to use supplied data-field instead of asking the user to enter it?

```javascript
const createSiteMutationDocument = gql`
    mutation createSite($input: createSiteInput!) {
        createSite(input: $input) {
            id
        }
    }
`;

ReactDOM.render(
    <ApplicationForm
        title="Todo Form"
        liveValidate={true}
        config={{
            mutation: {
                name: 'createSite',
                document: createSiteMutationDocument,
                variables: {input: {clientMutationId: "1"}}
            }
        }}
        data={{input:{clientMutationId: "2"}}}
    />
    , document.getElementById('root'));
```
I tried both `data` and `variables` without success

**Comment by wittydeveloper** [comment]
Hey @Richard87,

When you say:

> I tried both data and variables without success

Does that mean that the value `1` or `2` is not sent to the server or that the input still appears for this argument on the form?

Cheers,

**Comment by Richard87** [comment]
Hi, yeah, it was badly worded.

I want `clientMutationId` to be sent to the server, but not shown in the form...

**Comment by Richard87** [comment]
maby a property `skipFields` in combination with `values` taking a callback to add data, or a new field `presetData` that is included in the mutation, but not shown in the form?

**Comment by wittydeveloper** [comment]
Hey @Richard87 

normally this would work:

```ts
const createSiteMutationDocument = gql`
    mutation createSite($input: createSiteInput!) {
        createSite(input: $input) {
            id
        }
    }
`;

ReactDOM.render(
    <ApplicationForm
        title="Todo Form"
        liveValidate={true}
        config={{
            mutation: {
                name: 'createSite',
                document: createSiteMutationDocument,
                variables: {input: {clientMutationId: "1"}}
            },
            ignoreFields: ['input.clientMutationId']
        }}
        data={ {} }
    />
    , document.getElementById('root'));
```

`config.ignoreFields` is here to hide specific inputs/fields from the form.
See the [associated doc about config](https://github.com/wittydeveloper/react-apollo-form/wiki/Getting-started:-build-a-GraphQL-Form-in-5-minutes#propsconfig--apolloformconfig) 😉 

Cheers,

**Comment by Richard87** [comment]
Hmm, the submit button is disabled wjen ignoring `input.clientMutationId`, 
but it does work for hiding the input...

If I supply the "clientMutationId" as data as well, the submit button is activated, but the form fails client-side validation...

```
message:"is a required property"
name:"required"
params:{missingProperty: "clientMutationId"}
property:".input.clientMutationId"
stack:".input.clientMutationId is a required property"
```

Code:
```javascript
ReactDOM.render(
    <div className="container">
    <div className="col-md-6 col-md-offset-3"><ApplicationForm
        title="Todo Form"
        liveValidate={true}
        config={{
            mutation: {
                name: 'createSite',
                document: createSiteMutationDocument,
                variables: {input: {clientMutationId:"1"}}
            },
            ignoreFields: ['input.clientMutationId']
        }}
        data={{"input":{"clientMutationId": "1"}}}
    /></div>
    </div>
    , document.getElementById('root'));
```

**Comment by wittydeveloper** [comment]
Hey @Richard87,

I think you found a bug actually by trying to ignore a required field.
I'll take a look and try to fix it this week-end.

Sorry for the delay,

**Comment by wittydeveloper** [comment]
Hi @Richard87,

I've opened a pull request to fix your bug:
https://github.com/wittydeveloper/react-apollo-form/pull/15

I'll test it fully and deploy it to `npm` by the end of the week 🚢 

Thanks for your patience 🙇

**Comment by wittydeveloper** [comment]
@Richard87 fix published in the last version `0.1.6`

Let me know!
